docs: add comprehensive docstrings to PdfPipelineOptions

panwarnalini-hub opened the pull request:
## Summary
Adds comprehensive docstrings to `PdfPipelineOptions` class with detailed parameter documentation and production usage examples.

## Changes
- Added detailed class-level docstring to `PdfPipelineOptions`
- Documented all 9 parameters with explanations, defaults, and use cases
- Included 4 complete usage examples covering common scenarios
- Added production considerations (Databricks compatibility, timeout recommendations, OCR engine issues)
- Preserved existing inline comments and type hints

## Motivation
Addresses #2747 - improves developer experience by providing clear documentation for pipeline configuration options based on real production usage.

## Examples included
1. Basic digital PDF extraction (no OCR)
2. Scanned PDF with OCR and high-resolution images
3. Scientific papers with tables and formulas
4. Databricks-compatible configuration

Based on production experience with clinical trial document processing using Docling v2.

cau-git commented on docling/datamodel/pipeline_options.py:
```diff
@@ -399,7 +399,133 @@ class VlmExtractionPipelineOptions(PipelineOptions):
+                    InputFormat.PDF: PdfFormatOption(pipeline_options=options)
+                }
+            )
+
```
@panwarnalini-hub I suggest to remove this lengthy examples section on using the pipeline options with the DocumentConverter. They would fit well for general documentation and example codes, but not for the docstrings of the options type itself.

panwarnalini-hub replied:
Thanks for the suggestion! I've moved all the examples to docs/examples/pipeline_options_advanced.py. The docstrings now only contain brief descriptions with a cross-reference to the examples file.

panwarnalini-hub commented:
> Thanks @panwarnalini-hub for your effort on issue #2747 Please, consider the following observations:
> 
> * The attribute descriptions are accurate and relevant (excellent job!)
> * The code examples in the class docstrings are however excessive. We believe that this kind of examples are more appropriate in other sections like [examples](https://docling-project.github.io/docling/examples/). Could you please remove it from the docstrings?
> * I understand that many attributes were accidentally removed and then restored. Would it be possible to add the documentation of all the attributes? (e.g., `layout_options` , `ocr_batch_size`, ...)
> * Remove the inline documentation on the attribute declaration since they will become redundant with the docstrings (e.g., `  # True: perform table structure extraction`)
> * (small style comment): to be more aligned with google docstrings, I prefer the summary content right after the triple quotes, in a single line.

Thanks for the feedback! I've made the following changes:

- Removed all inline comments from attribute declarations
- Moved comprehensive examples to docs/examples/pipeline_options_advanced.py
- Added cross-reference in the docstring

The examples file now contains 10 different configuration scenarios, including production-grade patterns. Please let me know if any further adjustments are needed.

panwarnalini-hub commented:
> @panwarnalini-hub From the content of the docstring it looks like you would like to run Docling within Databricks. As @cau-git pointed out, the docstring might not be the proper place for it. But if are willing to contribute it (in another PR) I suggest you write some docs page on how to run Docling in Databricks and Spark.

I'd be happy to create a separate guide on running Docling in Databricks and Spark environments. I'll work on that as a follow-up PR once this one is merged. Thanks for the encouragement!

ceberam commented on docling/datamodel/pipeline_options.py:
```diff
@@ -106,159 +125,494 @@ class OcrOptions(BaseOptions):
-
-    # Dictionary to overwrite or pass-through additional parameters
-    rapidocr_params: Dict[str, Any] = Field(default_factory=dict)
-
```
Observation: the use of a default factory to prevent the issue of _mutable object as a default value for a function or method argument_ is not necessary here, since pydantic deals with this situation in base models.
https://docs.pydantic.dev/latest/concepts/fields/#mutable-default-values
